cognite/client/data_classes/filters.py

```diff
@@ -193,7 +193,10 @@ def _involved_filter_types(self) -> set[type[Filter]]:
         output = {type(self)}
         if isinstance(self, CompoundFilter):
             for filter_ in self._filters:
-                output.update(filter_._involved_filter_types())
+                if isinstance(filter_, Filter):
+                    output.update(filter_._involved_filter_types())
+                else:
+                    raise ValueError(f"Unknown filter type: {type(filter_)}. Expected Filter got {type(filter_)}")
```

Comment:
Should this rather be checked in `CompoundFilter.__init__`?
```suggestion
                    raise TypeError(f"Unknown filter type: {type(filter_)}")
```